flang/lib/Evaluate/intrinsics.cpp

```diff
@@ -949,6 +960,10 @@ static const IntrinsicInterface genericIntrinsicFunction[]{
     {"__builtin_ieee_support_nan",
         {{"x", AnyReal, Rank::elemental, Optionality::optional}},
         DefaultLogical},
+    {"__builtin_ieee_support_rounding",
+        {{"round_value", IeeeRoundType, Rank::scalar},
+            {"x", AnyReal, Rank::elemental, Optionality::optional}},
```

Comment:
Double check - Does this function description accommodate an optional **array** argument `X` in addition to a **scalar** argument `X`?  That is, is it valid to use implementation type `Rank::elemental` in this description for non-elemental , *inquiry* function `ieee_support_rounding` (and similar function descriptions.)?

Reply:
Yes.